shm/Player.cpp

```diff
@@ -0,0 +1,49 @@
+#include "Player.hpp"
+
+Player::Player(std::unique_ptr<Ship> &ship, int money, int availableSpace)
+    : ship_(std::move(ship)), money_(money), availableSpace_(calculateAvailableSpace())
+{}
+
+std::unique_ptr<Ship> &Player::getShip()
+{
+    return ship_;
+}
+
+int Player::getMoney()
+{
+    return money_;
+}
+
+size_t Player::getAvailableSpace()
+{
+    return availableSpace_;
+}
+
+size_t Player::getSpeed() const
+{
+    return ship_->getSpeed();
+}
+
+Cargo *Player::getCargo(size_t index) const
+{
+    return ship_->getCargo()[index];
```

Comment:
What if ptr is nullptr?
This is really unsafe. You have an empty vector at the beginning. So you access to a random place in memory which causes UB!